numpy/lib/npyio.py

```diff
@@ -85,8 +85,8 @@ def __init__(self, obj):
     def __getattribute__(self, key):
         try:
             return object.__getattribute__(self, '_obj')[key]
-        except KeyError:
-            raise AttributeError(key)
+        except KeyError as e:
+            raise AttributeError(key) from e
```

Comment:
This one would be better as the `from None` pattern, as is usually the case with `KeyError`

Reply:
I made this change as per your comment then did `git commit --amend` but when I tried `git push origin chanining-exceptions`, I got the following response:
```
Username for 'https://github.com': bijeshmohan
Password for 'https://bijeshmohan@github.com': 
To https://github.com/bijeshmohan/numpy.git
 ! [rejected]            chaining-exceptions -> chaining-exceptions (non-fast-forward)
error: failed to push some refs to 'https://github.com/bijeshmohan/numpy.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
I am sorry for the trouble again

Reply:
Use `git push origin chanining-exceptions -f`

Reply:
`git commit --amend` __changes__ the last commit. So now you have a commit in your local repo that cannot be pushed to the remote repo, since they conflict: they cannot be both present on the same branch without creating two heads
```
base ---- first commit
    \---- second commit
```
Best practice, especially as you get started, is to never `--amend`. If you would have committed without the `--amend`, you would have stacked the commits on top of eachother
```
base --- first commit --- second commit
```
But all is not lost. You can tell git to forget the first commit by "force pushing" to your repo
```
git push -f origin chanining-exceptions
```

Reply:
Thank you very much @eric-wieser. I think I just did that and it worked.

Reply:
Or you could `git pull`, then merge the heads with `git merge`
```
base ---- first commit ----merge commit
    \---- second commit ---/
```

Reply:
Thank you @mattip. Next time I will try this one!